File: 프로그래머스/lv1/12943. 콜라츠 추측/콜라츠 추측.cpp
#include <string>
#include <vector>

using namespace std;

int solution(int num)
{
    int answer = 0;
    long answer2 = num;

    while (answer2 != 1)
    {
        if (answer2 % 2 == 0)
        {
            answer2 /= 2;
            answer++;
        }
        else if (answer2 % 2 != 0)
        {
            answer2 *= 3;
            answer2 += 1;
            answer++;
        }
        if (answer >= 500)
            return -1;
    }

    return answer;
}

File: 프로그래머스/1/12943. 콜라츠 추측/콜라츠 추측.java
class Solution {
    public int solution(int num) {
        if (num == 1) return 0;
        
        int count = 0;
        while (count <= 500) {
            if (num % 2 == 1) {
                num = num * 3 + 1;
            } else {
                num /= 2;
            }
            count++;
            
            if (num == 1) return count;
        }
        
        return -1;
    }
}